rsconf/component/devbox.py

```diff
@@ -82,25 +81,21 @@ def internal_build_write(self):
         for k, v in z.secrets.items():
             self.install_abspath(v, z.host[k])
 
-    def _network(self, jc, z):
-        n = self.buildt.get_component("network")
-        z.ip, _ = n.ip_and_net_for_host(jc.rsconf_db.host)
-        z.ssh_port = jc.devbox.users[self.user_name]
-
     def _gen_host_and_identity_ssh_keys(self, jc):
         from rsconf import db
 
-        p = db.random_string()
+        f = db.secret_path(jc, _PASSWD_SECRET_JSON_F, visibility="host")
+        o = pkjson.load_any(f) if f.check() else PKDict()
         res = super()._gen_host_and_identity_ssh_keys(
-            jc, "devbox/" + self.user_name, visibility="host", password=p
+            jc,
+            "devbox/" + self.user_name,
```

Comment:
I liked your module_name concept. Maybe this could be part of the creation of the instance in the base class so that a component has a name and a module_name, and then overriden above when creating the child instances.

Reply:
I'm not sure I understand what you're looking for. Is this it 9d156a0 ?